integration-tests/spring-extension-support-test/src/test/java/com/example/application/AutoGridTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2023 Marco Collovati, Dario Götze
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.application;
+
+import java.util.List;
+
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.SelenideElement;
+import io.quarkus.test.junit.QuarkusTest;
+import org.junit.jupiter.api.Test;
+
+import com.github.mcollovati.quarkus.testing.AbstractTest;
+
+import static com.codeborne.selenide.CollectionCondition.size;
+import static com.codeborne.selenide.Selectors.byXpath;
+import static com.codeborne.selenide.Selectors.shadowCss;
+import static com.codeborne.selenide.Selectors.shadowDeepCss;
+import static com.codeborne.selenide.Selenide.$;
+import static com.codeborne.selenide.Selenide.$$;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@QuarkusTest
+class AutoGridTest extends AbstractTest {
+
+    private static final int ALL_ITEMS = 3;
+
+    @Override
+    protected String getTestUrl() {
+        return getBaseURL() + "auto-grid";
+    }
+
+    @Test
+    void autoGrid_gridIsDisplayed() {
+        openAndWait(() -> $("vaadin-grid"));
+        assertThat(collectColumnTexts(1)).containsExactlyElementsOf(List.of("Homer", "Jason", "Peter"));
+
+        SelenideElement filterRows =
+                $$(shadowCss("thead#header tr[part~=\"row\"]", "vaadin-grid")).last();
+        filterRows
+                .$$(shadowDeepCss("vaadin-text-field[placeholder=\"Filter...\"]"))
+                .shouldHave(size(2));
```

Comment:
What is checking this?

Reply:
Does it mean, there must be two filters visible ?

Reply:
Yes, there should be two text field for filtering the grid, one per filterable column.

Reply:
Is there another option to count these fields without looking at the placeholder ?

Reply:
The placeholder was the only thing I have found to be sure the textfield is used as a Grid filter